### ISSUE TITLE
Key Vault unzipping responses

### ISSUE THREAD

**Issue by H2RockyRoad** [microsoft/azure-spring-boot#921]
We have implemented azure keyvault secrets spring boot starter, but it has resulted in our rest responses coming back unzipped.

## Environment
- Spring boot starter:
  - key vault spring boot starter
  
- OS Type: MacOS
- Java version: 
  - 1.8
  

## Summary
When we added the dependency azure keyvault, our rest template responses are now coming back as ASCII text, when they used to come back GZipped.

## Reproduce steps
1) Add com.microsoft.azure:azure-keyvault-secrets-spring-boot-starter:2.2.0 to build.gradle 

## Expected Results
See POST response.body is GZipped

## Actual Results
See POST response.body is type ASCII

**Comment by yiliuTo** [comment]
Thanks for your reporing, could you describe how you use the key vault starter in your project or just add it as a dependency?

**Comment by H2RockyRoad** [comment]
The only thing we've done is add com.microsoft.azure:azure-keyvault-secrets-spring-boot-starter:2.2.0 to our build.gradle file. Before we got a chance to fully implement it, we noticed existing rest responses were no longer coming back GZipped.

**Comment by yiliuTo** [comment]
Hi @H2RockyRoad , could you provide a sample project or code snippet about the REST request/responses settings?

**Comment by moarychan** [comment]
Hi @H2RockyRoad , I will focus on this issue, could you please provide more details of the code structure,? I am checking the version you mentioned.
It's best if you can provide your demo project, I can reproduce your problem faster☺

**Comment by moarychan** [comment]
Hi @H2RockyRoad , 2.2.0 version found a problem. The spring boot 2.2.0 and azure-keyvault-secrets-starter 2.2.2 version combination work well,it is recommended that you upgrade the version of keyvault-secrets to 2.2.2 version. I added a rest api, check the Content-Encoding of response header  through the Postman, it has been compressed using gzip. This is the [sample ](https://github.com/moarychan/azure-starter/tree/master/key-vault/azure-spring-boot-sample-keyvault-secrets)I used for your reference.
